rest/config.go

```diff
@@ -842,6 +842,11 @@ func (sc *StartupConfig) validate() (errorMessages error) {
 			errorMessages = multierror.Append(errorMessages, fmt.Errorf("Cannot use supplied TLS key or cert when api.https.use_tls_client is false"))
 		}
 	}
+
+	if !base.IsEnterpriseEdition() && sc.API.EnableAdminAuthenticationPermissionsCheck != nil && *sc.API.EnableAdminAuthenticationPermissionsCheck {
+		errorMessages = multierror.Append(errorMessages, fmt.Errorf("enable_admin_authentication_permissions_check is only supported in enterprise edition"))
```

Comment:
Would it be better if the section was included where the config value is?
```suggestion
		errorMessages = multierror.Append(errorMessages, fmt.Errorf("api.enable_admin_authentication_permissions_check is only supported in enterprise edition"))
```

Reply:
Reason I omitted that was because that option is available in both legacy and  but you have a point... I think its probably fine to leave as is as a user would have to have explicitly set it to get this error so they probably would know where it is and wouldn't have to search for it....

You do have an interesting point tho